handlers/reader.go

```diff
@@ -9,42 +9,59 @@ import (
 	"net/http"
 
 	"github.com/openfaas/faas/gateway/requests"
+	"k8s.io/api/apps/v1beta2"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
-func getServiceList(functionNamespace string, clientset *kubernetes.Clientset) ([]requests.Function, error) {
-	functions := []requests.Function{}
+func getService(name, functionNamespace string, clientset *kubernetes.Clientset) (*requests.Function, error) {
+	deployment, err := clientset.AppsV1beta2().Deployments(functionNamespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
```

Comment:
I am not sure we want to hide the not found error here, especially because the function does return an error, I would expect an error on the not found.